modules/metastorage/src/main/java/org/apache/ignite/metastorage/internal/MetaStorageManager.java

```diff
@@ -75,77 +104,209 @@ public MetaStorageManager(
+    public @NotNull CompletableFuture<Void> removeAll(@NotNull Collection<Key> keys) {
+        return svc.removeAll(keys);
+    }
+
+    /**
+     * @see MetaStorageService#getAndRemoveAll(Collection)
+     */
+    public @NotNull CompletableFuture<Map<Key, Entry>> getAndRemoveAll(@NotNull Collection<Key> keys) {
+        return svc.getAndRemoveAll(keys);
+    }
+
+    /**
+     * @see MetaStorageService#invoke(Key, Condition, Operation, Operation)
      */
-    public CompletableFuture<Boolean> invoke(@NotNull Key key, @NotNull Condition condition,
+    public @NotNull CompletableFuture<Boolean> invoke(@NotNull Key key, @NotNull Condition cond,
         @NotNull Operation success, @NotNull Operation failure) {
-        return service.invoke(key, condition, success, failure);
+        return svc.invoke(key, cond, success, failure);
     }
 
     /**
-     * Gets a metastorage entry by key.
-     *
-     * @param key Key in metastorage.
-     * @param revision Metastorage update revision.
-     * @return Future.
+     * @see MetaStorageService#getAndInvoke(Key, Condition, Operation, Operation)
      */
-    public CompletableFuture<Entry> get(@NotNull Key key, long revision) {
-        return service.get(key, revision);
+    public @NotNull CompletableFuture<Entry> getAndInvoke(@NotNull Key key, @NotNull Condition cond,
+        @NotNull Operation success, @NotNull Operation failure) {
+        return svc.getAndInvoke(key, cond, success, failure);
     }
 
     /**
-     * Puts an entry to metastorage.
-     *
-     * @param key Key in metastorage.
-     * @param value Value bytes.
-     * @return Future.
+     * @see MetaStorageService#range(Key, Key, long)
      */
-    public CompletableFuture<Void> put(@NotNull Key key, @NotNull byte[] value) {
-        return service.put(key, value);
+    public @NotNull Cursor<Entry> range(@NotNull Key keyFrom, @Nullable Key keyTo, long revUpperBound) {
+        return svc.range(keyFrom, keyTo, revUpperBound);
     }
 
+    /**
+     * @see MetaStorageService#range(Key, Key)
+     */
+    public @NotNull Cursor<Entry> range(@NotNull Key keyFrom, @Nullable Key keyTo) {
+        return svc.range(keyFrom, keyTo);
+    }
 
-    public void registerWatch() {
+    /**
+     * @see MetaStorageService#compact()
+     */
+    public @NotNull CompletableFuture<Void> compact() {
+        return svc.compact();
+    }
 
+    /**
+     * Stop current batch of consolidated watches and register new one from current {@link WatchAggregator}.
+     *
+     * @return Ignite UUID of new consolidated watch
+     */
+    private CompletableFuture<IgniteUuid> updateWatches() {
+        igniteUuidFuture = igniteUuidFuture
+            .thenCompose(svc::stopWatch)
+            .thenCompose(v -> vaultManager.appliedRevision())
+            .thenCompose(r -> {
+                var watch = watchAggregator.watch(r, this::storeEntries);
+                return svc.watch(
+                    watch.keyCriterion().toRange().getKey(),
```

Comment:
Cosmetic: it seems to me, both keys that represent a `range` are "equivalent", So, it would be correct to use `get1()` and `get2()` instead of `key()` and `getValue()`, I think.